physical/file.go

```diff
@@ -121,19 +137,43 @@ func (b *FileBackend) Get(k string) (*Entry, error) {
 }
 
 func (b *FileBackend) Put(entry *Entry) error {
-	path, key := b.path(entry.Key)
+	if entry == nil {
+		return fmt.Errorf("nil entry")
+	}
+
+	var err error
+	basePath, fileName := b.path(entry.Key)
+
+	// New storage entries will have their file names base64 URL encoded. If a
+	// file with a non-encoded file name exists, it indicates that this is an
+	// update operation. To avoid duplication of storage entries, delete the
+	// old entry first.
```

Comment:
Done for most of it. I am not sure about the advantages of stating both versions and deleting it. What could go wrong if we try to delete the encoded one and if that fails, try deleting the old one? Is it that we'll not be able to detect the failure of deletion properly?